### PR TITLE
!Breaking Changes: Separate backend TRPC layer from frontend

### LINKED ISSUE #99: bug(web-app): handle if user does not have org
if a user dosnt have a default org (or any org) they will be redirected and get tons of errors

best place to handle this would be in the org redirect: `apps/web-app/pages/redirect.vue`

we have a "new org" form that would be a good candidate as a destination: `/apps/web-app/pages/[orgSlug]/settings/org/new.vue`
but we cant redirect here as its under the `[orgSlug]` folder and would send back to redirect
the new page would need to factor in the auth middleware to ensure users can navigate to it

### PR DESCRIPTION
**Pull request by BlankParticle**
## What does this PR do?

# Breaking Changes

This PR separates the backend layer from the nuxt webapp, before merging this PR, you should test everything extensively.
Also some environment variables have been changed, please check the `.env.example` file for the new variables.

Also Fixes #99

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [x] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Tested my code in a local environment
- [x] Commented on my code in hard-to-understand areas
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

File: .env.local.example
[the remark this answers is not in the document]
Comment (McPizza0) [reply .env.local.example:1]: completed

File: apps/platform/utils/auth/passkeys.ts [hunk 1]
```diff
@@ -115,12 +116,10 @@ async function generateAuthenticationOptions({
 
 async function verifyAuthenticationResponse({
   authenticationResponse,
-  authChallengeId,
-  expectedAllowedCredentials
+  authChallengeId
 }: {
   authenticationResponse: AuthenticationResponseJSON;
   authChallengeId: string;
-  expectedAllowedCredentials?: any;
```
Comment (McPizza0) [inline apps/platform/utils/auth/passkeys.ts:123]: we should revert this!

expectedCrendtials is used when we know the user that wants to login/verify - so we pass an array of acceptable authenticators that the UI will show

its used often for "re-auth" to verify the person taking the action is indeed the user (delete an org, set up billing etc...)

not used yet, but will be soon

File: apps/web-app/app/spa-loading-template.html
Comment (McPizza0) [inline apps/web-app/app/spa-loading-template.html:1]: why removed?

**Review by McPizza0 — CHANGES_REQUESTED** [review]
great work
left some comments
pushed some changes

File: apps/platform/utils/auth/passkeys.ts (hunk 1 above)
Comment (BlankParticle) [reply apps/platform/utils/auth/passkeys.ts:123]: sure, I will re-add it. I thought you left it out.

File: apps/web-app/app/spa-loading-template.html
Comment (BlankParticle) [reply apps/web-app/app/spa-loading-template.html:1]: I think I accidentally deleted it. didn't notice any difference as it barely makes any difference. I will update the file to better suit our needs.

File: apps/platform/trpc/routers/convoRouter/convoRouter.ts [hunk 2]
```diff
@@ -25,11 +25,15 @@ import {
 } from '@uninbox/database/schema';
 import { nanoId, nanoIdLength, zodSchemas } from '@uninbox/utils';
 import { TRPCError } from '@trpc/server';
-import type { JSONContent } from '@tiptap/vue-3';
+import type { AnyExtension, JSONContent } from '@tiptap/vue-3';
 import { generateText } from '@tiptap/core';
 import { generateHTML } from '@tiptap/html';
-import { tipTapExtensions } from '~/shared/editorConfig';
+import StarterKit from '@tiptap/starter-kit';
```
[the remark this answers is not in the document]
Comment (BlankParticle) [reply apps/platform/trpc/routers/convoRouter/convoRouter.ts:31]: Sure, I will make a separate package called `@uninbox/tiptap` for that.

**Comment by BlankParticle** [comment]
You should do the changes as I will not be available for sometime, So if you do it and merge the backend then I will be able to pickup on new stuff when I have time.

File: apps/platform/trpc/routers/convoRouter/convoRouter.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (McPizza0) [reply apps/platform/trpc/routers/convoRouter/convoRouter.ts:31]: will open new issue for this

**Review by McPizza0 — APPROVED** [review]

**Pull request merged**